common/egov-idgen/src/main/java/org/egov/id/config/SequenceResetService.java

```diff
@@ -0,0 +1,100 @@
+package org.egov.id.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+
+@Service
+@Slf4j
+public class SequenceResetService {
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @Value("${dristi.kollam.court.id}")
+    private String KOLLAM_COURT_ID;
+
+    private static final String ADVOCATE_SEQUENCE = "SEQ_REG_ADV";
+    private static final String CLERK_SEQUENCE = "SEQ_REG_CLERK";
+    private static final String FILING_SEQUENCE = "SEQ_FILING_NUMBER";
+    private static final String CNR_SEQUENCE = "SEQ_CNR_";
+    private static final String COURT_CASE_SEQUENCE = "SEQ_CCST_";
+    private static final String CMP_SEQUENCE = "SEQ_CMP_";
+
+    private static final String TIME_ZONE = "Asia/Kolkata";
+
+    // This runs at midnight on December 31st. One second before transitioning to january 1st every year
+    //59 59 23 31 12 *
+
+    @Scheduled(cron = "59 59 23 31 12 *", zone = TIME_ZONE)
+    public void resetAdvocateSequence() {
+        try {
+            String sql = "ALTER SEQUENCE " + ADVOCATE_SEQUENCE + " RESTART WITH 1;";
```

Comment:
logging curval is not possible because it return value of last executed NEXTVAL command in that session. calling curval might thrwoing this exception "currval of sequence "seq_reg_adv" is not yet defined in this session"